docs/how-to/hip_rtc.rst

```diff
@@ -0,0 +1,727 @@
+    to :cpp:func:`hiprtcGetLoweredName` to extract the lowered name.
+
+  * The correct sequence of calls is : :cpp:func:`hiprtcAddNameExpression`,
+    :cpp:func:`hiprtcCreateProgram`, :cpp:func:`hiprtcGetLoweredName`,
+    :cpp:func:`hiprtcDestroyProgram`.
+
+  * The lowered names must be fetched using :cpp:func:`hiprtcGetLoweredName`
+    only after the HIPRTC program has been compiled, and before it has been
+    destroyed.
+
+Example
+-------------------------------------------------------------------------------
+
+Kernel containing various definitions ``__global__`` functions/function
+templates and ``__device__/__constant__`` variables can be stored in a string.
+
+.. code-block:: cpp
+    
+  static constexpr const char gpu_program[] {
+  R"(
+  __device__ int V1; // set from host code
+  static __global__ void f1(int *result) { *result = V1 + 10; }
+  namespace N1 {
+  namespace N2 {
+  __constant__ int V2; // set from host code
+  __global__ void f2(int *result) { *result = V2 + 20; }
+  }
+  }
+  template<typename T>
+  __global__ void f3(int *result) { *result = sizeof(T); }
+  )"};
+
+:cpp:func:`hiprtcAddNameExpression` is called with various name expressions
+referring to the address of ``__global__`` functions and
+``__device__/__constant__`` variables.
+
+.. code-block:: cpp
+
+  kernel_name_vec.push_back("&f1");
+  kernel_name_vec.push_back("N1::N2::f2");
+  kernel_name_vec.push_back("f3<int>");
+  for (auto&& x : kernel_name_vec) hiprtcAddNameExpression(prog, x.c_str());
+  variable_name_vec.push_back("&V1");
+  variable_name_vec.push_back("&N1::N2::V2");
+  for (auto&& x : variable_name_vec) hiprtcAddNameExpression(prog, x.c_str());
+
+After which, the program is compiled using :cpp:func:`hiprtcCompileProgram` and
+the generated binary is loaded using :cpp:func:`hipModuleLoadData`. And the
+mangled names can be fetched using :cpp:func:`hirtcGetLoweredName`.
+
+.. code-block:: cpp
+
+  for (decltype(variable_name_vec.size()) i = 0; i != variable_name_vec.size(); ++i) {
+    const char* name;
+    hiprtcGetLoweredName(prog, variable_name_vec[i].c_str(), &name);
+  }
+
+.. code-block:: cpp
+
+  for (decltype(kernel_name_vec.size()) i = 0; i != kernel_name_vec.size(); ++i) {
+    const char* name;
+    hiprtcGetLoweredName(prog, kernel_name_vec[i].c_str(), &name);
+  }
+
+The mangled name of the variables are used to look up the variable in the module
+and update its value.
+
+.. code-block:: cpp
+
+  hipDeviceptr_t variable_addr;
+  size_t bytes{};
+  hipModuleGetGlobal(&variable_addr, &bytes, module, name);
+  hipMemcpyHtoD(variable_addr, &initial_value, sizeof(initial_value));
+
+
+Finally, the mangled name of the kernel is used to launch it using the
+``hipModule`` APIs.
+
+.. code-block:: cpp
+
+  hipFunction_t kernel;
+  hipModuleGetFunction(&kernel, module, name);
+  hipModuleLaunchKernel(kernel, 1, 1, 1, 1, 1, 1, 0, nullptr, nullptr, config);
+
+Please have a look at ``hiprtcGetLoweredName.cpp`` for the detailed example.
```

Comment:
I found a reference for this at https://github.com/ROCm/HIP-Common/blob/develop/tests/src/hiprtc/hiprtcGetLoweredName.cpp, but i am not sure this is the one you want to point to. In any case, I think some kind of reference here would be helpful.